census/ui_app/QuestionForm.jsx

```diff
@@ -186,20 +233,66 @@ const QuestionForm = React.createClass({
       if (this.getSchemaForId(q.id) === undefined) {
         console.warn('No schema defined for Question with id: ' + q.id);
       }
-      return (
-        <QuestionField ref={q.id}
-                       key={q.id}
-                       id={q.id}
-                       visibleProps={this.getVisiblePropsForId(q.id)}
-                       value={q.value}
-                       onChange={this.onFieldChange}
-                       label={this.getLabelForId(q.id)}
-                       position={this.getPositionForId(q.id)}
-                       instructions={this.getValueForId(q.id, 'description')}
-                       placeholder={this.getValueForId(q.id, 'placeholder')}>
-          {this.getValueForId(q.id, 'text')}
-        </QuestionField>
-      );
+      let questionNode = '';
+      switch (this.getValueForId(q.id, 'type')) {
```

Comment:
I've committed a solution to a separate branch based on what you suggest: https://github.com/okfn/opendatacensus/commit/5cf128c82f9b7c80c6e36a490c5521feabd80f33